### PR TITLE
BaSP-MR-11: Subscription put and delete

### PR DESCRIPTION
**Pull request by IgnacioGalceran**
Added Subscription schema.
Added put and delete controllers for subscription.
Added validations to update subscriptions

### CODE REVIEW

File: src/validations/subscription.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const Joi = require('joi');
+
+const validate = (req, res, next) => {
+  const subscriptionValidation = Joi.object({
+    classes: Joi.string().regex(/^[0-9a-fA-F]{24}$/),
```
[the remark this answers is not in the document]
Comment (Eliezer-Alberto) [reply src/validations/subscription.js:5]: You can add a custom function to validate if it is an object id..
`Const id = 1`

[the remark this answers is not in the document]
Comment (IgnacioGalceran) [reply src/validations/subscription.js]: done! :D

File: src/controllers/subscription.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    return applyResponse(res, 404, 'Id is invalid', undefined, true);
+  }
+  return Subscription.findByIdAndDelete(id)
+    .then((result) => {
+      if (!result) {
+        return applyResponse(res, 404, `Subscription with id: ${id} was not found`, undefined, true);
+      }
+      return applyResponse(res, 200, `Subscription with id ${id} was deleted`, result, false);
+    })
+    .catch((error) => res.status(400).json({ msg: error.message, error: true }));
```
Comment (juanlantermo) [inline src/controllers/subscription.js:74]: Hi Ignacio! If I'm not mistaken at the time of the catch, the error is from the server and the status number should be 500

File: src/controllers/subscription.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        })
+        .then((result) => applyResponse(
+          res,
+          200,
+          `Subscription with id: ${id} was updated successfully`,
+          result,
+          false,
+        ));
+    })
+    .catch((error) => res.status(400).json({ msg: error.message, error: true }));
```
Comment (juanlantermo) [inline src/controllers/subscription.js:59]: Here I also think I would be in the same situation

Comment (IgnacioGalceran) [reply src/controllers/subscription.js:59]: ah right! i'll change it

Comment (IgnacioGalceran) [reply src/controllers/subscription.js]: changed

**Comment by Eliezer-Alberto** [comment]
I think everything is Ok!

**Review by Dannylez — APPROVED** [review]
🚀

**Review by Eliezer-Alberto — APPROVED** [review]
It is all fine for me..

**Review by nailaghigo — APPROVED** [review]

**Pull request merged**